schema-engine/sql-introspection-tests/tests/postgres/constraints.rs

```diff
@@ -4,7 +4,464 @@ use indoc::indoc;
+          is_locked Int?
+        }
+
+        model notifications {
+          notification_id  Int           @id
+          created_at       DateTime?     @default(now()) @db.Timestamptz(6)
+          subscription_id  Int
+          return_values    Json?
+          block_time       DateTime      @db.Timestamptz(6)
+          transaction_hash String        @db.VarChar(255)
+          block            BigInt
+          sent             Boolean       @default(false)
+          subscriptions    subscriptions @relation(fields: [subscription_id], references: [subscription_id], onDelete: Cascade, onUpdate: NoAction, map: "notifications_subscription_id_foreign")
+
+          @@index([subscription_id], map: "notifications_subscription_id_index")
+        }
+
+        model subscriptions {
+          subscription_id Int             @id
+          user_id         Int
+          eventsource_id  Int
+          created_at      DateTime?       @default(now()) @db.Timestamptz(6)
+          join_block      BigInt
+          notifications   notifications[]
+          eventsources    eventsources    @relation(fields: [eventsource_id], references: [eventsource_id], onDelete: NoAction, onUpdate: NoAction, map: "subscriptions_eventsource_id_foreign")
+          users           users           @relation(fields: [user_id], references: [user_id], onDelete: Cascade, onUpdate: NoAction, map: "subscriptions_user_id_foreign")
+
+          @@unique([user_id, eventsource_id], map: "subscriptions_user_id_eventsource_id_unique")
+          @@index([eventsource_id], map: "subscriptions_eventsource_id_index")
+          @@index([user_id], map: "subscriptions_user_id_index")
+        }
+
+        /// This table contains check constraints and requires additional setup for migrations. Visit https://pris.ly/d/postgres-check-constraints for more info.
+        model tokens {
+          token_id       Int       @id
+          created_at     DateTime? @default(now()) @db.Timestamptz(6)
+          user_id        Int
+          token_scope    String?   @default("MAGICLINK")
+          valid          Boolean   @default(true)
+          some_new_field String?   @db.VarChar(255)
+          users          users     @relation(fields: [user_id], references: [user_id], onDelete: Cascade, onUpdate: NoAction, map: "tokens_user_id_foreign")
+
+          @@index([user_id], map: "tokens_user_id_index")
+        }
+
+        model users {
+          user_id       Int             @id
+          created_at    DateTime?       @default(now()) @db.Timestamptz(6)
+          email         String          @unique(map: "users_email_unique") @db.VarChar(255)
+          verified      Boolean         @default(false)
+          subscriptions subscriptions[]
+          tokens        tokens[]
+
+          @@index([email], map: "users_email_index")
+        }
+    "#]];
+
+    api.expect_datamodel(&schema).await;
+
+    // ensure the introspected schema is valid
+    psl::parse_schema(schema.data()).unwrap();
```

Comment:
We don't really need to validate this here. We test these in the datamodel renderer...